src/main/java/com/google/cloud/spanner/pgadapter/statements/BackendConnection.java

```diff
@@ -1313,26 +1312,15 @@ int executeStatementsInBatch(int fromIndex) {
     return index - fromIndex;
   }
 
-  /**
-   * Extracts the update count for a list of DDL statements. It could be that the DdlExecutor
-   * decided to skip some DDL statements. This is indicated by the executor returning a {@link
-   * NotExecuted} result for that statement. The result that we return to the client should still
-   * indicate that the statement was 'executed'.
-   */
+  /** Extracts the update count for a list of DDL statements. */
   static long[] extractDdlUpdateCounts(
       List<StatementResult> statementResults, long[] returnedUpdateCounts) {
     int index = 0;
     int successfullyExecutedCount = 0;
     while (index < returnedUpdateCounts.length
         && returnedUpdateCounts[index] == 1
         && successfullyExecutedCount < statementResults.size()) {
-      if (!(statementResults.get(successfullyExecutedCount) instanceof NotExecuted)) {
-        index++;
-      }
-      successfullyExecutedCount++;
-    }
-    while (successfullyExecutedCount < statementResults.size()
-        && statementResults.get(successfullyExecutedCount) instanceof NotExecuted) {
+      index++;
```

Comment:
It seems that both index and successfullyExecutedCount will have the same value. Do we need both the fields?

Reply:
Good point, they are indeed now duplicates of each other, as we will send all statements to the backend. Removed the separate `index` variable.